stetho/src/main/java/com/facebook/stetho/inspector/network/NetworkEventReporterImpl.java

```diff
@@ -152,10 +152,52 @@ public void responseHeadersReceived(InspectorResponse response) {
           getResourceTypeHelper().determineResourceType(contentType) :
           Page.ResourceType.OTHER;
       receivedParams.response = responseJSON;
+      AsyncPrettyPrinter asyncPrettyPrinter =
+          initAsyncPrettyPrinterForResponse(response, peerManager);
+      modifyReceivedParamsForAsyncPrettyPrinter(receivedParams, asyncPrettyPrinter);
       peerManager.sendNotificationToPeers("Network.responseReceived", receivedParams);
     }
   }
 
+  @Nullable
+   static AsyncPrettyPrinter initAsyncPrettyPrinterForResponse(
+      InspectorResponse response,
+      NetworkPeerManager peerManager) {
+    AsyncPrettyPrinterRegistry registry = peerManager.getAsyncPrettyPrinterRegistry();
+    AsyncPrettyPrinter asyncPrettyPrinter = createPrettyPrinterForResponse(response, registry);
+    if (asyncPrettyPrinter != null) {
+      peerManager.getResponseBodyFileManager().associateAsyncPrettyPrinterWithId(
+          response.requestId(),
+          asyncPrettyPrinter);
+    }
+     return asyncPrettyPrinter;
+  }
+
+  private void modifyReceivedParamsForAsyncPrettyPrinter(
```

Comment:
This function has some readability challenges.  First, it claims to modify parameters but it really only conditionally does that and the condition isn't clear at a high level.  Second, there's no reason to allow the original assignment to proceed since you know that the pretty printer can override it at the very time that it was previously determined.  You can express this logic much more concisely as:

```java
private static Page.ResourceType determineResourceType(...) {
  if (asyncPrettyPrinter != null) {
    return ...
  } else {
    return ... getResourceTypeHelper().determineResourceType(...);
  }
}
```